recipes/tree-sitter/all/conanfile.py

```diff
@@ -0,0 +1,60 @@
+import os
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import shutil
+
+required_conan_version = ">=1.29.1"
+
+class TreeSitterConan(ConanFile):
+    name = "tree-sitter"
+    description = "Tree-sitter is a parser generator tool and an incremental parsing library. It can build a concrete syntax tree for a source file and efficiently update the syntax tree as the source file is edited."
+    topics = ("parser")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://tree-sitter.github.io/tree-sitter"
+    license = "MIT"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"fPIC": [True, False],
+               "shared": [True, False]}
+    default_options = {"fPIC": True,
+                       "shared": False}
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def requirements(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("tree-sitter is not support on {}.".format(self.settings.os))
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("{}-{}".format(self.name, self.version), self._source_subfolder)
+
+    def _configure_autotools(self):
+        if not self._autotools:
+            self._autotools = AutoToolsBuildEnvironment(self)
+
+        return self._autotools
+
+    def build(self):
+        autotools = self._configure_autotools()
+        with tools.chdir(self._source_subfolder):
+            tools.replace_in_file("Makefile", "PREFIX ?= /usr/local", "PREFIX ?= {}".format(self.package_folder))
+            autotools.make()
+
+    def package(self):
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+
+        autotools = self._configure_autotools()
+        with tools.chdir(self._source_subfolder):
+            autotools.install()
```

Comment:
```suggestion
        autotools.install(args=["PREFIX={}".format(self.package_folder)])
```

Reply:
It works! What kind of magic do your use??=) The only moment is that we still need `with tools.chdir(self._source_subfolder):`
because make file is in source sub dir

Reply:
Your move prefix to install command, exclude lib prefix in cpp info and it works? Why?)) I really don't understand=)

Reply:
I think this stackoverflow question sums it up:
https://stackoverflow.com/questions/448910/what-is-the-difference-between-the-gnu-makefile-variable-assignments-a

Reply:
I think your problem was that, for shared libraries, multiple files (with shared links) were installed.
e.g.
```
lib/libtree-sitter.so
lib/libtree-sitter.so.1
lib/libtree-sitter.so.1.2
```
And your deletion by `*.so` would not remove the actual shared library, only one shared link.

Reply:
It make sense, but that is not the hardest problem I facet, I cannot fix link error if lib name in info cpp was without extension